src/main/java/net/neoforged/neoforge/data/event/GatherDataEvent.java

```diff
@@ -139,4 +145,67 @@ public void runAll() {
             });
         }
     }
+
+    public boolean shouldRun(Dist dist) {
+        return switch (dist) {
+            case CLIENT -> includeClient();
+            case DEDICATED_SERVER -> includeServer();
+        };
+    }
+
+    public void addProvider(boolean run, DataProvider provider) {
```

Comment:
I am concerned that all these helpers are just called `addProviders`, there was an issue in the past regarding ambiguous lambdas with some of the Neo helpers as `DataProvider` itself is a functional interface.

Reply:
yes, @gigaherz mentioned that.
if i got it right this is caused because java does not enforce generics at runtime.
we don't have that problem here because there are no generics in play.

Reply:
Have you verified its not an issue in this case? I can assure you that `DataProvider` was the class in question that caused problems in the past, and it does not use generics.

Reply:
I did not run into any issues while writing this pr and other mods (using a wrapper class) but I can not cover all use cases.
It might be that something regarding this changed with Java versions.
Do you remember where this problem was discussed earlier so I can reproduce the exact circumstances?